mypy_json_report.py

```diff
@@ -12,15 +12,49 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import argparse
+import enum
 import json
 import sys
 from collections import Counter, defaultdict
 from dataclasses import dataclass
 from typing import Counter as CounterType, Dict, Iterator
 
 
+class ErrorCodes(enum.IntEnum):
+    DEPRECATED = 1
+
+
 def main() -> None:
-    print(produce_errors_report(sys.stdin))
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(title="subcommand")
+
+    parser.set_defaults(func=_deprecated_command)
+
+    parse_parser = subparsers.add_parser(
+        "parse", help="Transform Mypy output into JSON."
+    )
+
+    parse_parser.set_defaults(func=_parser_command)
+
+    args = sys.argv[1:]
+    parsed = parser.parse_args(args)
+    parsed.func(parsed)
+
+
+def _parser_command(args: object) -> None:
+    report_errors()
+
+
+def _deprecated_command(args: object) -> None:
+    print("A subcommand is required. Pass --help for usage info.")
+    sys.exit(ErrorCodes.DEPRECATED)
+
+
+def report_errors() -> None:
+    errors = parse_errors_report(sys.stdin)
+    error_json = json.dumps(errors, sort_keys=True, indent=2)
+    print(error_json)
```

Comment:
**Note:** I've isolated this so that I can alter the data source / output file in a later PR.